projects/batfish-common-protocol/src/main/java/org/batfish/storage/Storage.java

```diff
@@ -0,0 +1,14 @@
+package org.batfish.storage;
+
+import java.nio.file.Path;
+import java.util.stream.Stream;
+
+/** Common storage APIs */
+public interface Storage {
+
+  Path getContainer(String containerName);
```

Comment:
The `Storage` abstractions do not know about files. They need to be flexible and adaptable to any backend storage system – database, file system, cloud storage, etc. Instead of `Path`, we should be returning things like `Container`.

Everywhere the Coordinator or Batfish worker assume files we need to insert a layer of indirection – so that things are addressed by (e.g.) container name , container name + testrig name, etc.